### PR TITLE
Integrate ruff format

### PR DESCRIPTION
**Pull request by Mic92**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: .bbtravis.yml [hunk 1]
```diff
@@ -181,6 +181,12 @@ script:
       export PATH=/tmp/bbvenv/bin/:$PATH
       isort --check -df `git ls-files |grep '.py$'`
 
+  - title: ruff-format
+    condition: TESTS == "ruff"
```
Comment (Mic92) [inline .bbtravis.yml:185]: Don't know what this condition does. I just copied it from the other examples.

File: .bbtravis.yml [hunk 2]
```diff
@@ -181,6 +181,12 @@ script:
       export PATH=/tmp/bbvenv/bin/:$PATH
       isort --check -df `git ls-files |grep '.py$'`
```
Comment (Mic92) [inline .bbtravis.yml:182]: We can also replace pylint/isort/flake8 with ruff as it integrates the same rulesets.
Than we no longer have competing linters.

**Comment by Mic92** [comment]
Since there is still a conflict between ruff and isort, I will replace isort with ruff's own sorting in the next PR.

File: .bbtravis.yml (hunk 2 above)
Comment (Mic92) [reply .bbtravis.yml]: started in https://github.com/buildbot/buildbot/pull/7326

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,7 @@
+[tool.ruff]
+line-length = 88
+target-version = "py38"
```
Comment (Mic92) [inline pyproject.toml:3]: What's the lowest version still supported here?

**Comment by Mic92** [comment]
Since I am getting a lot of these "line-too-long" errors now, would it be acceptable to disable this lint and just let `ruff` make sure that long lines are broken down? Most of these warnings come from strings.

**Comment by p12tic** [comment]
I would like to use `--skip-string-normalization` black option because most formatting changes are for quotes (which don't really matter when considering the benefits of using black/ruff. It doesn't seem ruff has this option, does it?

**Comment by Mic92** [comment]
Looks like it was added here: https://github.com/astral-sh/ruff/pull/8822

**Comment by Mic92** [comment]
ruff > [v0.1.8](https://github.com/astral-sh/ruff/releases/tag/v0.1.8)

**Comment by Mic92** [comment]
Just let me know your preference and I can update this pull request.

File: pyproject.toml (hunk 3 above)
Comment (p12tic) [reply pyproject.toml:3]: 3.8 is correct. For worker minimum is 3.4.

Comment (p12tic) [reply pyproject.toml:3]: line-length should be 100.

**Comment by p12tic** [comment]
> Just let me know your preference and I can update this pull request.

Let's enable `--skip-string-normalization` and convert. I don't have anything large planned that could need a backport to 3.x, so there will be little inconvenience.

File: pyproject.toml (hunk 3 above)
Comment (Mic92) [reply pyproject.toml:3]: Done.

**Comment by Mic92** [comment]
The big ruff format is now down automatically. You can reproduce by running:

```
ruff format .
ruff lint --fix --select "ISC" .
isort .
git commit -am 're-format with ruff'
```

**Comment by Mic92** [comment]
Finally all linters seems to be happy now!

**Comment by p12tic** [comment]
@Mic92 FYI in the end import order didn't need to be fixed. This was probably caused by some confusion over what's created by ruff reformat and what are later manual changes.

**Pull request merged**